alibi_detect/saving/loading.py

```diff
@@ -607,3 +532,27 @@ def _replace(cfg: dict, orig: Optional[str], new: Optional[str]) -> dict:
         elif isinstance(v, dict):
             _replace(v, orig, new)
     return cfg
+
+
+def _prepend_cfg_filepaths(cfg: dict, prepend_dir: Path):
+    """
+    Recursively traverse through a nested dictionary and prepend a directory to any filepaths.
+
+    Parameters
+    ----------
+    cfg
+        The dictionary.
+    prepend_dir
+        The filepath to prepend to any filepaths in the dictionary.
+
+    Returns
+    -------
+    The updated config dictionary.
+    """
+    for k, v in cfg.items():
+        if isinstance(v, str):
+            v = prepend_dir.joinpath(Path(v))
+            if v.is_file() or v.is_dir():  # Update if prepending config_dir made config value a real filepath
```

Comment:
Do you iterate through recursively and if it's a string you prepend the directory and see if it exists in the file system? I'm trying to think if there's some way this might backfire?

Reply:
Yes exactly that. Then only update the string in the config (with the new prepended file path) if it is a file or dir. 

The two ways I can think of which it might backfire are:
1. If a config contains a field which a `str` but shouldn't be a filepath (e.g. a regsitry str), whose string is such that when prepending the config filepath to it the new string is actually a filepath. This field would then incorrectly be updated. I can't think of a realistic example of where this would happen, but it is theoretically possible.
2. The bigger disadvantage is that if a filepath has been spec'd in the config but it is missing/incorrect, it won't be updated with the prepended path. The `FileNotFoundError` error raised later would then be confusing if it prints out the problem directory. The easiest thing to do is probs just go through and check no `FileNotFoundError`'s print out the given directory...

Reply:
Following up on this, I've actually decided not to remove the filepath included in `FileNotFoundError`'s such as that in `saving.tensorflow._loading.load_model`:

```python
raise FileNotFoundError(f'No .h5 file found in {model_dir}.')
```

Although `model_dir` wouldn't have the config file dir prepended to it (see 2. above), the filepath would still match that in `config.toml`, which might actually be a nice feature to aid the user's debugging...

@mauicv shout if you disagree, otherwise think we're good to go!